lib/hooks/useHoliday.ts

```diff
@@ -0,0 +1,67 @@
+import { useEffect, useState } from "react"
+import { ApiGetResponse } from "../types/hooks"
+import { NEXT_PUBLIC_GOOGLE_APIKEY } from "../conf"
+
+type Holiday = {
+    date: string
+    description: string
+    summary: string
+}
+
+interface IData extends Holiday {
+    start: {
+        date: string
+    }
+}
+
+function parseEvents(data: { items: IData[] }): Holiday[] {
+    const { items } = data
+    const holidays = []
+    for (const item of items) {
+        const { date } = item.start
+        const { description } = item
+        const { summary } = item
+        holidays.push({ date, description, summary })
+    }
+    return holidays
+}
+
+export const useHoliday = (): ApiGetResponse<Holiday[]> => {
+    const [response, setResponse] = useState<ApiGetResponse<Holiday[]>>({
+        isSuccess: false,
+        isLoading: true,
+        isError: false,
+    })
+
+    useEffect(() => {
+        const fetchData = async () => {
+            try {
+                const apiResponse = await fetch(
+                    // eslint-disable-next-line max-len
+                    `https://www.googleapis.com/calendar/v3/calendars/fi.finnish%23holiday%40group.v.calendar.google.com/events?key=${NEXT_PUBLIC_GOOGLE_APIKEY}`
```

Comment:
You can get rid of the eslint disable by declaring the url as a top level const e.g.

```ts
const GOOGLE_API_URL = 'https://www.googleapis.com/calendar/v3/calendars/fi.finnish%23holiday%40group.v.calendar.google.com/events'
```
and then in the method call
```ts
fetch(
    `${GOOGLE_API_URL}?key=${NEXT_PUBLIC_GOOGLE_APIKEY}`
)
```